src/main/java/core/basesyntax/strategy/impl/OperationStrategyImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.model.Operation;
+import core.basesyntax.strategy.OperationHandler;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.Map;
+
+public class OperationStrategyImpl implements OperationStrategy {
+    private final Map<Operation, OperationHandler> operationHandlers;
+
+    public OperationStrategyImpl(Map<Operation, OperationHandler> operationHandlers) {
+        this.operationHandlers = operationHandlers;
+    }
+
+    @Override
+    public OperationHandler getHandler(Operation operation) {
+        return operationHandlers.get(operation);
```

Comment:
Consider adding a check to handle cases where the requested operation does not have a corresponding handler in the map. This will prevent potential `NullPointerException` and provide a more informative error message or fallback behavior.